common/script/helpers.js

```diff
@@ -0,0 +1,32 @@
+import {
+  isEmpty,
+  max,
+  reduce,
+} from 'lodash';
+
+import uuid from 'uuid';
+
+export function $w (s) {
+  return s.split(' ');
+}
+
+/*
+  Reflists are arrays, but stored as objects. Mongoose has a helluvatime working with arrays (the main problem for our
+  syncing issues) - so the goal is to move away from arrays to objects, since mongoose can reference elements by ID
+  no problem. To maintain sorting, we use this helper function:
+ */
+
+export function refPush (reflist, item) {
+  item.sort = isEmpty(reflist) ? 0 : max(reflist, 'sort').sort + 1;
```

Comment:
the only thing I don't like here is that we're adding the `sort` property not only to the item pushed inside the refPush but also changing the original `item` object, it's safer if we `_.clone` the object before